specs/workload/Workload.md

```diff
@@ -0,0 +1,207 @@
+
+* Enumerate installed workloads
+* Retrieve properties of a workload
+* Remove installed packages and related resources comprising a workload
+
+## 3.1. API Structure
+
+## 3.2. Workload Definition via appxmanifest.xml
+
+A workload has the following properties:
+
+* Id -- programmatic identifier
+* Rank -- prioritization for selection if multiple workloads have the same id
+* DisplayName -- localizable name for user/human consumption
+* Handler -- WinRT object providing operations manipulating the workload
+* Supported Actions -- Operation(s) a  handler can perform
+
+An MSIX package can define one or more workloads. The following manifest schema is used:
+
+```xml
+<Package>
+  ...
+  <Properties>
+    <workload:Workloads>
+      <Workload Id="id" Rank="rank" DisplayName="dn">
+        <Actions ActivatableClassId="acid">
+          <action/>
+          ...
+        </Actions>
+      </Workload>
+      ...
+    </Workload:Workloads>
+  </Properties>
+  ...
+</Package>
+```
+
+The following validations apply:
+
+* Id (id)
+  * String length: 1-255
+  * Allowed characters: [A-Za-z0-9_.]
+  * Comparison: Case-Insensitive (locale-invariant)
+  * Occurrence: Required (minOccurs=maxOccurs=1)
+  * Validation: Unique across all workloads in a package
+* Rank (rank)
+  * Int32
+  * Occurrence: Required (minOccurs=maxOccurs=1)
+* DisplayName (dn)
+  * [Standard `DisplayName` definition](https://learn.microsoft.com/en-us/uwp/schemas/appxpackage/uapmanifestschema/element-displayname)
+  * Occurrence: Required (minOccurs=maxOccurs=1)
+* ActivatableClassId (acid)
+  * [Standard `ActivatableClassId` definition](https://learn.microsoft.com/en-us/uwp/schemas/appxpackage/uapmanifestschema/element-activatableclass)
+  * Occurrence: Optional (minOccurs=0, maxOccurs=1)
+* Actions
+  * Occurrence: Optional (minOccurs=0, maxOccurs=1)
+* actions
+  * The actions supported by this handler.
+  * Occurrence: Required (minOccurs=maxOccurs=1).
+  * NOTE: Currently only <Remove> is supported but other actions will be added in the future e.g. if a Repair action is supported expressing support for both Remove and Repair would be
+
+```xml
+        <Actions ActivatableClassId="acid">
+          <Remove/>
+          <Repair/>
+        </Actions>
+```
+
+### 3.2.1. Workload 'Id' and Collisions
+
+If packages are registered with multiple workloads containing the same `Id` then the workload with
+the highest `Rank` is used. If multiple workloads have the same `Id` and `Rank` then they are
+assumed to be equivalent and one will be used.
+
+NOTE: This is the same collision management algorithm as used by Undocked Deployment Extension
+Handlers (DEHs).
+
+# 4. Examples
+
+## 4.1. Remove all workloads
+
+Fabrikam app removing all workloads.
+
+```c#
+namespace RemoveAllWorkloads
+{
+    static class Program
+    {
+        static void Main()
+        {
+            var workloadManager = WorkloadManager.GetDefault();
```

Comment:
Manager looks to be a deprecated term in the implementation. Should it still exist here?

Reply:
Good catch. Fixed